src/hotspot/share/adlc/adlArena.cpp

```diff
@@ -129,28 +130,33 @@ void* AdlArena::grow( size_t x ) {
 //------------------------------calloc-----------------------------------------
 // Allocate zeroed storage in AdlArena
 void *AdlArena::Acalloc( size_t items, size_t x ) {
+  assert(items <= SIZE_MAX / x, "overflow");
   size_t z = items*x;   // Total size needed
   void *ptr = Amalloc(z);       // Get space
   memset( ptr, 0, z );          // Zap space
   return ptr;                   // Return space
 }
 
 //------------------------------realloc----------------------------------------
+size_t pointer_delta(const void *left, const void *right) {
```

Comment:
Do we want to assert left >= right here?

Reply:
It's currently only used in one place, where we know that's true.  OTOH, an assert doesn't hurt.
Instead of this helper function, we could just use`(size_t)(_max - c_old)` inline (we can be
confident the difference won't exceed the `ptrdiff_t` range here), reducing the lines of code by
a little bit.  The benefit from the helper is having the Arealloc code (nearly? completely?) identical
here and in memory/arena.cpp.  I don't have a strong opinion either way.

Reply:
The motivation was to keep the two arenas consistent. To that end, I've now added an assert and made the function static.